packages/shorebird_cli/lib/src/commands/release_new/release_new_command.dart

```diff
@@ -0,0 +1,420 @@
+Use `shorebird flutter versions list` to list available versions.
+''',
+        );
+        exit(ExitCode.software.code);
+      }
+
+      return revision;
+    }
+
+    return shorebirdEnv.flutterRevision;
+  }
+
+  /// Asserts that a release with version [version] can be released using
+  /// flutter revision [flutterRevision]. If a release has already been
+  /// published with the given [version] for the platform [releasePlatform], or
+  /// if a release already exists with [version] but was compiled with a
+  /// different Flutter revision, an error will be thrown.
+  Future<void> assertVersionIsReleasable({
+    required String version,
+    required String flutterRevision,
+    required ReleasePlatform releasePlatform,
+  }) async {
+    final existingRelease = await codePushClientWrapper.maybeGetRelease(
+      appId: appId,
+      releaseVersion: version,
+    );
+
+    if (existingRelease != null) {
+      codePushClientWrapper.ensureReleaseIsNotActive(
+        release: existingRelease,
+        platform: releasePlatform,
+      );
+
+      // All artifacts associated with a given release must be built
+      // with the same Flutter revision.
+      if (existingRelease.flutterRevision != flutterRevision) {
+        // All artifacts associated with a given release must be built
+        // with the same Flutter revision.
+        logger
+          ..err('''
+${styleBold.wrap(lightRed.wrap('A release with version $version already exists but was built using a different Flutter revision.'))}
+''')
+          ..info('''
+
+  Existing release built with: ${lightCyan.wrap(existingRelease.flutterRevision)}
+  Current release built with: ${lightCyan.wrap(flutterRevision)}
+
+${styleBold.wrap(lightRed.wrap('All platforms for a given release must be built using the same Flutter revision.'))}
+
+To resolve this issue, you can:
+  * Re-run the release command with "${lightCyan.wrap('--flutter-version=${existingRelease.flutterRevision}')}".
+  * Delete the existing release and re-run the release command with the desired Flutter version.
+  * Bump the release version and re-run the release command with the desired Flutter version.''');
+        exit(ExitCode.software.code);
+      }
+    }
+  }
+
+  /// Prints a confirmation prompt with details about the release to be created.
+  /// If the user confirms, the release will be created. If the user cancels,
+  /// the command will exit with a success code. When running in a headless
+  /// or CI environment, this prompt will print but will not wait for user
+  /// confirmation.
+  Future<void> confirmCreateRelease({
+    required AppMetadata app,
+    required String releaseVersion,
+    required String flutterVersion,
+    required ReleasePlatform releasePlatform,
+  }) async {
+    final flutterVersionString = await shorebirdFlutter.getVersionAndRevision();
+    // TODO(bryanoltman): include archs in the summary for android
```

Comment:
Good idea, maybe we should address this TODO before landing? Feels like a important piece of info